components/supervisor/frontend/src/shared/frontend-dashboard-service.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { IDEFrontendDashboardService } from "@gitpod/gitpod-protocol/lib/frontend-dashboard-service";
+import { RemoteTrackMessage } from "@gitpod/gitpod-protocol/lib/analytics";
+import { Emitter } from "@gitpod/gitpod-protocol/lib/util/event";
+
+export class FrontendDashboardServiceClient implements IDEFrontendDashboardService.IClient {
+    public latestStatus!: IDEFrontendDashboardService.Status;
+    public sessionID!: string;
+
+    private readonly onDidChangeEmitter = new Emitter<IDEFrontendDashboardService.Status>();
+    readonly onStatusUpdate = this.onDidChangeEmitter.event;
+
+    private resolveInit!: () => void;
+    private initPromise = new Promise<void>((resolve) => (this.resolveInit = resolve));
+
+    constructor(private serverWindow: Window) {
+        console.log("===========new.FrontendDashboardServiceClient", serverWindow);
+        window.addEventListener("message", (event: MessageEvent) => {
```

Comment:
I remember that we would have check origin checks, ie loading screen will only accept messages originated from the top window, and top window will only accept messages originated from the loading screen origin. 

I’m not sure why we added it there. I think the browser has somehow distributed messaging architecture and prevent other windows to try to call to another window ie cross workspace window calls a check was added.

Reply:
It's recommended to add them, because `postMessage` is a cross-origin method and we don't want the details to end up in iframes we don't control. 

Some bits from the MDN doc [[1](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage#security_concerns)]:
> always verify the sender's identity using the origin and possibly source properties. Any window (including, for example, http://evil.example.com) can send a message to any other window, and you have no guarantees that an unknown sender will not send malicious messages.


> Always specify an exact target origin, not *, when you use postMessage to send data to other windows. A malicious site can change the location of the window without your knowledge, and therefore it can intercept the data sent using postMessage.